### PR TITLE
DEPR: NDFrame.set_axis inplace defaults to false #27525

### PR DESCRIPTION
**Pull request by ghost**
https://github.com/pandas-dev/pandas/blob/76247c142893c710e970c4cf8a25d73121aa5a2b/pandas/core/generic.py#L594-L598

has been there since <del>#20164</del> #16994 (issue was #14636), part of 0.21.0.

With discussion of plans to deprecate similar functionality from set_index in #24046, it's time to make sure `set_axis` conforms to the rest of pandas in this.

### CODE REVIEW

File: pandas/core/generic.py [hunk 1]
```diff
@@ -671,14 +657,7 @@ def set_axis(self, labels, axis=0, inplace=None):
...
         if inplace is None:
-            warnings.warn(
-                "set_axis currently defaults to operating inplace.\nThis "
-                "will change in a future version of pandas, use "
-                "inplace=True to avoid this warning.",
-                FutureWarning,
-                stacklevel=2,
-            )
-            inplace = True
+            raise ValueError("`inplace` cannot have None as a value.")
```
Comment (WillAyd) [inline pandas/core/generic.py]: I think you can just delete this altogether - don't think we do this anywhere else (though correct if wrong)

File: doc/source/whatsnew/v1.0.0.rst [hunk 2]
```diff
@@ -61,7 +61,7 @@ Deprecations
 Removal of prior version deprecations/changes
 ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
 - Removed the previously deprecated :meth:`Series.get_value`, :meth:`Series.set_value`, :meth:`DataFrame.get_value`, :meth:`DataFrame.set_value` (:issue:`17739`)
--
+- Changed the the default value of `inplace` in of :meth:`DataFrame.set_index` and :meth:`Series.set_index`. It now defaults to False (:issue:``)
```
Comment (WillAyd) [inline doc/source/whatsnew/v1.0.0.rst]: set_axis

**Review by WillAyd — CHANGES_REQUESTED** [review]
looks pretty good minor comments

**Comment by WillAyd** [comment]
In case anyone is looking for it I think you meant to reference #16994 (the OP just points to a documentation update)

File: pandas/core/generic.py (hunk 1 above)
Comment (pilkibun) [reply pandas/core/generic.py]: No, I think you're right

File: doc/source/whatsnew/v1.0.0.rst (hunk 2 above)
Comment (pilkibun) [reply doc/source/whatsnew/v1.0.0.rst]: thanks

File: doc/source/whatsnew/v1.0.0.rst [hunk 3]
```diff
@@ -61,7 +61,7 @@ Deprecations
 Removal of prior version deprecations/changes
 ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
 - Removed the previously deprecated :meth:`Series.get_value`, :meth:`Series.set_value`, :meth:`DataFrame.get_value`, :meth:`DataFrame.set_value` (:issue:`17739`)
--
+- Changed the the default value of `inplace` in :meth:`DataFrame.set_index` and :meth:`Series.set_axis`. It now defaults to False (:issue:``)
```
Comment (WillAyd) [inline doc/source/whatsnew/v1.0.0.rst]: Think accidentally deleted issue number

File: doc/source/whatsnew/v1.0.0.rst [hunk 4]
```diff
@@ -191,6 +191,10 @@ ExtensionArray
 -
 
 
+Other
```
Comment (WillAyd) [inline doc/source/whatsnew/v1.0.0.rst]: Can you delete this while changing whatsnew?

**Review by WillAyd — COMMENTED** [review]
Minor whatsnew quibbles but lgtm otherwise

File: doc/source/whatsnew/v1.0.0.rst (hunk 3 above)
Comment (pilkibun) [reply doc/source/whatsnew/v1.0.0.rst]: I wrote this before I opened the PR...

**Comment by pilkibun** [comment]
Thanks for the review

**Comment by jreback** [comment]
thanks @pilkibun

**Comment by pilkibun** [comment]
cc @toobaz,  you had some sentiments relating this to the `set_index` vs. `set_axis` debate  in https://github.com/pandas-dev/pandas/pull/16994#issue-130885544 in relation to `set_index` vs. `set_axis`. I'm not what your position was, but It sounds like two years ago you were against both the little-endians *and* the big-endians ;)

... Can we haz some docs now?

**Comment by toobaz** [comment]
@pilkibun I think I'm missing something, as I don't understand.

But related to the comment you link, I guess it is probably time to remove this warning I had introduced...

https://github.com/pandas-dev/pandas/blob/37f29e5bf967d76d76a05e84b5c9b14c9bc66f23/pandas/core/generic.py#L650

(and the subsequent reassignment).

**Comment by pilkibun** [comment]
> @pilkibun I think I'm missing something, as I don't understand.

You objected to "publicizing" `set_axis` in the docs because of its defects and also that changing the default to `inplace=False` (this PR) is necessary before it can become officially recommended. 

Now that that's done, I hope you'll push for making a decision on `set_index` vs. `set_axis`, so that we can finally document what the "official" method is.

**Comment by toobaz** [comment]
OK, now I see where you're going.

The defects of ``set_axis`` are, as you saw, a thing of the past, and they are/were entirely unrelated to ``set_index``. You can just go on with a nice docs PR "publicizing" ``set_axis`` if you like.

The API contains a lot of "official" methods to do a lot of different things.

Please stop provoking.

**Pull request merged**